Documentation/Administration/Deployment/Automated/Index.rst

```diff
@@ -0,0 +1,195 @@
+    `vendor/bin/typo3 database:updateschema` if :composer:`helhum/typo3-console`
+    is installed.
+*   **Cache Clearing**: Clear TYPO3 caches
+    (`vendor/bin/typo3 cache:flush`).
+
+..  _ci-cd-environment-configuration:
+
+Environment configuration
+--------------------------
+
+Manage environment-specific settings using
+`.env / dotenv files <https://docs.typo3.org/permalink/t3coreapi:environment-dotenv>`_
+or `Plain PHP configuration files <https://docs.typo3.org/permalink/t3coreapi:environment-phpconfig>`_.
+
+..  _ci-cd-typical-tools:
+
+Typical CI/CD Tools Used
+------------------------
+
+CI/CD Platforms
+    `GitHub Actions <https://docs.typo3.org/permalink/t3coreapi:ci-cd-github>`_,
+    `GitLab CI <https://docs.typo3.org/permalink/t3coreapi:ci-cd-gitlab>`_,
+    `Jenkins <https://www.jenkins.io/>`_, or `CircleCI <https://circleci.com/>`_
+
+Code Quality
+    `PHP-CS-Fixer <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-cs>`_,
+    `PHPstan <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-phpstan>`_,
+    and `TypoScript-Lint <https://github.com/martin-helmich/typo3-typoscript-lint>`_
+
+Testing
+    `PHPUnit <https://phpunit.de/>`_ with the `TYPO3 Testing Framework <https://docs.typo3.org/permalink/t3coreapi:testing-writing-unit>`_
+
+Build Tools
+    Composer and Webpack, Gulp, or Vite
+
+Deployment Tools
+    `Deployer <https://docs.typo3.org/permalink/t3coreapi:deployment-deployer>`_,
+    `Rsync <https://docs.typo3.org/permalink/t3coreapi:deployment-rsync>`_, Helm, Ansible, GitOps
+
+Docker Support
+    DDEV, Docker Compose
+
+..  _ci-cd-plattforms:
+
+CI/CD Platforms
+===============
+
+..  _ci-cd-gitlab:
+
+Using GitLab CI
+---------------
+
+The `Official GitLab template for TYPO3 <https://docs.typo3.org/permalink/t3start:gitlab-template>`_
+provides a predefined `.gitlab-ci.yml <https://gitlab.com/gitlab-org/project-templates/typo3-distribution/-/blob/main/.gitlab-ci.yml>`_
+and a `Deployer recipe <https://docs.typo3.org/permalink/t3coreapi:deployer-gitlab>`_
+that you can customize to your needs.
+
+Even if you already set up your project you can find valuable examples there
+
+..  _ci-cd-github:
+
+Using GitHub Actions
+--------------------
+
+.. code-block:: yaml
+
+    name: TYPO3 CI/CD Pipeline
+
+    on:
+      push:
+        branches: [ "main" ]
+
+    jobs:
+      build:
+        runs-on: ubuntu-latest
+
+        steps:
+          - uses: actions/checkout@v4
+
+          - name: Set up PHP
+            uses: shivammathur/setup-php@v2
+            with:
+              php-version: '8.1'
+
+          - name: Install Dependencies
+            run: composer install --no-dev --prefer-dist
+
+          - name: Lint PHP
+            run: find . -name "*.php" -exec php -l {} \;
+
+          - name: Run PHPStan
+            run: vendor/bin/phpstan analyse
+
+          - name: Run PHPUnit Tests
+            run: vendor/bin/phpunit
+
+          - name: Deploy (Example)
```

Comment:
thinking about it, i would split the CI part form the deployment as both should not happen at the same time.

Reply:
Would you like to provide a better example file? I have never done this so it is quite theoretic